### PR TITLE
Camera Alerts For Engineering

### PR DESCRIPTION
**Pull request by bathosbathtime**
This PR lets engineering know what cameras are disabled.  This currently uses the alarm wire to add and remove cameras from the station alert console as well as cameras that start disabled.  This only includes cameras on zlevel one.  This does include cameras disabled roundstart.  Do note that you will need to snip and mend the alarm wire to remove it from the alerts if it was disabled roundstart.

Compiled and tested locally.

:cl:
 * rscadd: Added cameras to the engineering alert console if they send an alarm.
 * rscadd: Cameras disabled at round start send an alarm.

### CODE REVIEW

**Comment by ShiftyRail** [comment]
>This also includes cameras that are not on the station network (like central command) and will need to be fixed.

You'd need to change `triggerCameraAlarm` so that it only dials 911 to cameras on the same z-level.

**Comment by ghost** [comment]
Also please change it so roundstart camera breakages are listed on the alerts, please. Giving AIs blind spots is one thing but making them impossible to track down is just shit

**Comment by bathosbathtime** [comment]
That should do it, it now only sends an alarm for zlevel 1 cameras.

**Comment by ghost** [comment]
I still think that making the roundstart broken cameras add themselves to the alert computers is really important for quality of life

**Comment by bathosbathtime** [comment]
Yes, that is the point of this PR.

**Comment by ghost** [comment]
Are we 100 % sure it works ? Not to be a dong but I tried the solution in your code and it did not index in the Silicon alerts panel

File: code/game/machinery/camera/camera.dm [hunk 1]
```diff
@@ -402,6 +406,9 @@ var/list/camera_messages = list()
 	var/area/this_area = get_area(src)
 	for(var/mob/living/silicon/S in mob_list)
 		S.cancelAlarm("Camera", this_area, src)
+	if(z == 1)
```
Comment (Kurfursten) [inline code/game/machinery/camera/camera.dm]: STATION_Z (for the one above too)

**Comment by bathosbathtime** [comment]
I have added the requested change to fix the station check, but still need to fix silicons not getting the roundstart camera alerts.  Right now I believe it to be caused by the fact that the alerts are happening before the silicon exists in the game.
Left window is the engineer cyborg's alert verb, the right window is the station alert console.
siliconbug

**Comment by Intigracy** [comment]
I would suggest making the computers and the silicons share a datum full of the ongoing camera alerts rather than updating them on an individual mob level as the alerts happen.

**Comment by Kurfursten** [comment]
You can request this reopened if you want to work on it

**Pull request closed without merging**